Frontend/src/Components/Faults/FaultsView.js

```diff
@@ -1,13 +1,334 @@
-import { Heading, HStack, Text, VStack } from "@chakra-ui/react";
+import { Box, Tooltip, Image, SimpleGrid } from "@chakra-ui/react";
+import MPPTContactorImage from "./Images/MPPT Contactor.png";
+import LowContactorImage from "./Images/Low Contactor.png";
+import MotorControllerContactorImage from "./Images/Motor Controller Contactor.png";
+import DoorImage from "./Images/Door.png";
+import BatteryFailsafeImage from "./Images/Battery Failsafe.png";
+import BPSFaultImage from "./Images/BPS Fault.png";
+import EStopImage from "./Images/E-Stop.png";
+import IMDStatusImage from "./Images/IMD Status.png";
+import CrashImage from "./Images/Crash.png";
+import HighCurrentImage from "./Images/High Current.png";
+import HighVoltageImage from "./Images/High Voltage.png";
+import LowCurrentImage from "./Images/Low Current.png";
+import LowVoltageImage from "./Images/Low Voltage.png";
+import LowBatteryImage from "./Images/Low Battery.png";
+import MCUCheckImage from "./Images/MCU Check.png";
+import MPPTCurrentImage from "./Images/MPPT Current.png";
+import PowerWarningImage from "./Images/Power Warning.png";
+import BMSInputVoltageImage from "./Images/BMS Input Voltage.png";
+import PhysicalConnectionImage from "./Images/Physical Connection Lost.png";
+import WirelessCommsLostImage from "./Images/Wireless Comms Lost.png";
+import HighTemperatureImage from "./Images/High Temperature.png";
 
 export default function Faults(props) {
+    const getEStopString = () => {
+        let eStopStrings = [];
+
+        if(props.data?.battery_eStop[0]) {
+            eStopStrings.push("Battery");
+        }
+        if(props.data?.driver_eStop[0]) {
+            eStopStrings.push("Driver");
+        }
+        if(props.data?.external_eStop[0]) {
+            eStopStrings.push("External");
+        }
+
+        if(eStopStrings.length > 0) {
+            if(eStopStrings.length > 2) {
+                return eStopStrings[0] + ", " + eStopStrings[1].toLowerCase() + ", and " + eStopStrings[2].toLowerCase() + " E-stops were pressed";
+            } else if(eStopStrings.length > 1) {
+                return eStopStrings[0] + " and " + eStopStrings[1].toLowerCase() + " E-stops were pressed";
+            } else {
+                return eStopStrings[0] + " E-stop was pressed";
+            }
+        }
```

Comment:
```suggestion
        if(eStopStrings.length > 0) {
            buf = "eStopStrings[0] ";
            for(let i=i; i < eStopStrings.length; i++) { // Starting at the 2nd index because first index isn't prepped with a comma
                buf += `, and ${eStopStrings[i].toLowerCase()} `;
               }
               buf += "E-stop was pressed";
               return buf;
           }
```

That way if you add more estop strings you don't have to rewrite this function